### PR TITLE
css_compress: use placeholder to keep quoted strings

### LINKED ISSUE #2517: Some CSS selectors broken during CSS compression
The following CSS rule `[style="width: 60%;"]{background-color:red;}` added in `conf/userstyle.css` is transformed into `[style="width:60%;"]{background-color:red;}` (without the space in the CSS selector) in the webpage's `css.php` file.

A temporary way to get around that is to uncheck the 'compress' option in the configuration options. 

I'm on 2018-04-22a "Greebo".

### PR DESCRIPTION
**Pull request by phy25**
This fixes #2517.

The idea mainly comes from https://github.com/matthiasmullie/minify/blob/97f118c4c745c7c8e47207f2daf3bab13ca65404/src/Minify.php#L343; I also removed the quote detection code in `css_onelinecomment_cb()` because it's a duplicate of this.

### CODE REVIEW

**Comment by splitbrain** [comment]
I'm unsure about the placeholders. Would this break if someone uses a variable named `STR1` in their code?

**Comment by phy25** [comment]
Looking back, I just realized that my implementation is bad. Thanks @splitbrain!

What this does: `"doku"` > `STR1` > `"doku"`

[What minify does](https://github.com/matthiasmullie/minify/blob/97f118c4c745c7c8e47207f2daf3bab13ca65404/src/Minify.php#L343):  `"doku"` > `"1"` > `"doku"`

Will definitely fix this!

**Comment by phy25** [comment]
About the new commit: I did replace STRN with a new placeholder in the previous implementation so there is no conflict. However what minify does is more elegant, so this commit implements it.

Rebased due to tests totally not working. Double checked tests error and they are all about #2841. Once #2850 is merged and this is rebased again, tests should be green.

File: _test/tests/lib/exe/css_css_compress.test.php [hunk 1]
```diff
@@ -138,6 +138,13 @@ function test_data() {
         $this->assertEquals($expect, css_compress($input));
     }
 
+    function test_quotes() {
```
[the remark this answers is not in the document]
Comment (phy25) [reply _test/tests/lib/exe/css_css_compress.test.php]: I agree, it should be clearer. Polished it a little bit.

**Comment by micgro42** [comment]
Mh, leading/trailing whitespace of the css script _should_ not cause problems when it is removed, I hope

**Comment by phy25** [comment]
We are [doing this on JavaScript](https://github.com/splitbrain/dokuwiki/blob/546e4fa1a966866d8357d585a86872ceba169e8c/lib/exe/js.php#L487) already.

Looks like clean-css is doing this as well: https://github.com/jakubpawlowicz/clean-css/blob/844e152cfd9d2b83ac51624847d051c964f9abfa/lib/clean.js#L124

**Pull request merged**